x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TransportSqlQueryAction.java

```diff
@@ -12,37 +12,32 @@
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.inject.Inject;
 import org.elasticsearch.common.io.stream.Writeable;
-import org.elasticsearch.common.metrics.CounterMetric;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.watcher.common.stats.Counters;
 import org.elasticsearch.xpack.sql.action.SqlQueryAction;
 import org.elasticsearch.xpack.sql.action.SqlQueryRequest;
 import org.elasticsearch.xpack.sql.action.SqlQueryResponse;
 import org.elasticsearch.xpack.sql.execution.PlanExecutor;
 import org.elasticsearch.xpack.sql.proto.ColumnInfo;
 import org.elasticsearch.xpack.sql.proto.Mode;
-import org.elasticsearch.xpack.sql.proto.RestClient;
 import org.elasticsearch.xpack.sql.session.Configuration;
 import org.elasticsearch.xpack.sql.session.Cursors;
 import org.elasticsearch.xpack.sql.session.RowSet;
 import org.elasticsearch.xpack.sql.session.SchemaRowSet;
-import org.elasticsearch.xpack.sql.stats.Counters;
-import org.elasticsearch.xpack.sql.stats.Metric;
+import org.elasticsearch.xpack.sql.stats.Metrics;
+import org.elasticsearch.xpack.sql.stats.QueryMetric;
 import org.elasticsearch.xpack.sql.type.Schema;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
-import java.util.Map.Entry;
-import java.util.concurrent.ConcurrentHashMap;
 
 import static java.util.Collections.unmodifiableList;
 
 public class TransportSqlQueryAction extends HandledTransportAction<SqlQueryRequest, SqlQueryResponse> {
     private final PlanExecutor planExecutor;
     private final SqlLicenseChecker sqlLicenseChecker;
-    private static final Map<Metric, Map<String, CounterMetric>> opsByTypeMetrics = new ConcurrentHashMap<>();
-    private final Map<String, CounterMetric> featuresMetrics = new ConcurrentHashMap<>();
+    private static final Metrics metrics = new Metrics();
```

Comment:
Why is the metric created in the transport and not inside `PlanExecutor`. Further more why is it static (what's the point of setting it as an instance).
There's only one `PlanExecutor` which is shared across all transports - doing the initialization in the transport means that if translate is called instead of query, the metric won't be inside `PlanExecutor`.
Further more transports are auxiliary to the executor and thus should just get the final Metric from the executor without any setting which implies that `PlanExecutor` can exist in an inconsistent state.

Reply:
The Metrics instance was static because it needed to be accessed in the static `operation()` method. Also, translate operations are not managed in Metrics, they are incremented in a separate transport action in a much simpler way.
I've moved the Metrics to the `PlanExecutor`.